bundles/org.openhab.binding.melcloud/src/main/java/org/openhab/binding/melcloud/internal/handler/MelCloudDeviceHandler.java

```diff
@@ -0,0 +1,311 @@
+                break;
+            case CHANNEL_VANE_HORIZONTAL:
+                cmdtoSend.setVaneHorizontal(((DecimalType) command).intValue());
+                cmdtoSend.setEffectiveFlags(EFFECTIVE_FLAG_VANE_HORIZONTAL);
+                break;
+            default:
+                logger.debug("Read only or unknown channel {}, skip update", channelUID);
+        }
+
+        if (cmdtoSend.getEffectiveFlags() > 0) {
+            cmdtoSend.setHasPendingCommand(true);
+            cmdtoSend.setDeviceID(config.deviceID);
+            try {
+                DeviceStatus newDeviceStatus = melCloudHandler.sendDeviceStatus(cmdtoSend);
+                updateChannels(newDeviceStatus);
+            } catch (MelCloudCommException e) {
+                logger.warn("Command '{}' to channel '{}' failed, reason {}. ", command, channelUID, e.getMessage(), e);
+            }
+        } else {
+            logger.debug("Nothing to send");
+        }
+    }
+
+    private DeviceStatus getDeviceStatusCopy(DeviceStatus deviceStatus) {
+        DeviceStatus copy = new DeviceStatus();
+        synchronized (this) {
+            copy.setPower(deviceStatus.getPower());
+            copy.setOperationMode(deviceStatus.getOperationMode());
+            copy.setSetTemperature(deviceStatus.getSetTemperature());
+            copy.setSetFanSpeed(deviceStatus.getSetFanSpeed());
+            copy.setVaneVertical(deviceStatus.getVaneVertical());
+            copy.setVaneHorizontal(deviceStatus.getVaneHorizontal());
+            copy.setEffectiveFlags(deviceStatus.getEffectiveFlags());
+            copy.setHasPendingCommand(deviceStatus.getHasPendingCommand());
+            copy.setDeviceID(deviceStatus.getDeviceID());
+        }
+        return copy;
+    }
+
+    private void startAutomaticRefresh() {
+        if (refreshTask == null || refreshTask.isCancelled()) {
+            Runnable runnable = () -> getDeviceDataAndUpdateChannels();
+            refreshTask = scheduler.scheduleWithFixedDelay(runnable, 1, config.pollingInterval, TimeUnit.SECONDS);
+        }
+    }
+
+    private void getDeviceDataAndUpdateChannels() {
+        if (melCloudHandler.isConnected()) {
+            logger.debug("Update device '{}' channels", getThing().getThingTypeUID());
+            try {
+                DeviceStatus newDeviceStatus = melCloudHandler.fetchDeviceStatus(config.deviceID,
+                        Optional.ofNullable(config.buildingID));
+                updateChannels(newDeviceStatus);
+            } catch (MelCloudCommException e) {
+                logger.debug("Error occurred during device '{}' polling, reason {}. ", getThing().getThingTypeUID(),
+                        e.getMessage(), e);
+            }
+        } else {
+            logger.debug("Connection to MELCloud is not open, skip periodic update");
+        }
+    }
+
+    private synchronized void updateChannels(DeviceStatus newDeviceStatus) {
```

Comment:
This seems a bit to much synchronization: `synchronized` on the method and in the method.